meson.build

```
@@ -47,7 +48,7 @@ endif
 
 # Detect for version-script support
 ld_has_version_script = compiler.has_link_argument(
-    '-Wl,-version-script,@0@'.format(meson.source_root() / 'build-aux' / 'test.map')
+    '-Wl,-version-script,@0@'.format(meson.global_source_root() / 'build-aux' / 'test.map')
```

```suggestion
    '-Wl,-version-script,@0@'.format(meson.project_source_root() / 'build-aux' / 'test.map')
```

The two functions are equivalent except that global returns incorrect information if libdatrie is built as a subproject of another project -- it will then refer to the source root of the other project instead of libdatrie.